### DISCUSSION TITLE
Option to clear all images from a page

### DISCUSSION THREAD

**Discussion by Soremwar** [py-pdf/pypdf#2743] [Q&A] [answered]
## Explanation

I'd like to replace the contents of a page with multiple images for one with a single background image (preserving the page text), however since the image list is read only this is not possible

## Code Example

```python
from pypdf import PdfReader, PdfWriter

input_pdf = PdfReader(path)
output_pdf = PdfWriter(clone_from=input_pdf)

page = output_pdf.pages[0]
page.images.clear()
page.images.append(bg_image)
```

## Alternative

An option to clone a page leaving the images out

**Comment by pubpub-zz** [comment, marked as answer]
you have 2 options:
a) you know which image is the one to replace:
eg: 
```
from PIL import Image
(...)
page.image[0].replace(Image.open("image.jpg"))
```
b) you delete all images, then create new pdf using pillow , and merge pages placing the new one behind
```
from PIL import Image
(...)
page.remove_objects_from_page(page,ObjectDeletionFlag.XOBJECT_IMAGES)
img = Image.open("image.jpg")
b= BytesIO()
img.save(b,"pdf")
img_pdf = PdfReader(b)
img_page = img_pdf.pages[0]
page.merge_page(img_page,over=False)
```

PS: I convert this as a discussion as this not an issue
